src/main/java/movies/spring/data/neo4j/movies/MovieService.java

```diff
@@ -17,60 +23,89 @@
-
-				var targetIndex = nodes.size();
-				nodes.add(movie);
-
-				record.get("actors").asList(v -> v.asString()).forEach(name -> {
-					var actor = Map.of("label", "actor", "title", name);
-
-					int sourceIndex;
-					if (nodes.contains(actor)) {
-						sourceIndex = nodes.indexOf(actor);
-					} else {
-						nodes.add(actor);
-						sourceIndex = nodes.size() - 1;
-					}
-					links.add(Map.of("source", sourceIndex, "target", targetIndex));
-				});
-			});
-		}
-		return Map.of("nodes", nodes, "links", links);
-	}
+    private final MovieRepository movieRepository;
+
+    private final Driver driver;
+
+    private final String database;
+
+    MovieService(MovieRepository movieRepository, Driver driver, @Value("${spring.data.neo4j.database}") String database) {
+        this.movieRepository = movieRepository;
+        this.driver = driver;
+        this.database = database;
+    }
+
+    public Optional<MovieCast> findMovieByTitle(String title) {
+        return movieRepository.findOneByTitle(title)
+                .map(this::getCast);
+    }
+
+    public List<Movie> findAllMoviesByTitleLike(String title) {
+        return movieRepository.findAllByTitleLikeIgnoreCase(title);
+    }
+
+    /**
+     * This is an example of when you might want to use the pure driver in case you have no need for mapping at all, neither in the
+     * form of the way the {@link org.springframework.data.neo4j.core.Neo4jClient} allows and not in form of entities.
+     *
+     * @return A representation D3.js can handle
+     */
+    public Map<String, List<Object>> fetchMovieGraph() {
+
+        var nodes = new ArrayList<>();
+        var links = new ArrayList<>();
+
+        try (Session session = driver.session(forDatabase(database))) {
+            var records = session.readTransaction(tx -> tx.run(""
+                    + " MATCH (m:Movie) <- [r:ACTED_IN] - (p:Person)"
+                    + " WITH m, p ORDER BY m.title, p.name"
+                    + " RETURN m.title AS movie, collect(p.name) AS actors"
+            ).list());
+            records.forEach(record -> {
+                var movie = Map.of("label", "movie", "title", record.get("movie").asString());
+
+                var targetIndex = nodes.size();
+                nodes.add(movie);
+
+                record.get("actors").asList(v -> v.asString()).forEach(name -> {
+                    var actor = Map.of("label", "actor", "title", name);
+
+                    int sourceIndex;
+                    if (nodes.contains(actor)) {
+                        sourceIndex = nodes.indexOf(actor);
+                    } else {
+                        nodes.add(actor);
+                        sourceIndex = nodes.size() - 1;
+                    }
+                    links.add(Map.of("source", sourceIndex, "target", targetIndex));
+                });
+            });
+        }
+        return Map.of("nodes", nodes, "links", links);
+    }
+
+    private MovieCast getCast(Movie movie) {
```

Comment:
This is not super elegant. Just wanted to get to a state where it works first.
The plan is to remove this and replace it with a custom query and projection.